### PR TITLE
Updated login flow to fix provider selection when uid/password given.

### PR DESCRIPTION
**Pull request by ArmandGrillet**

### CODE REVIEW

**Comment by ArmandGrillet** [comment]
@mesosphere-ci run integration tests

File: pkg/prompt/prompt.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func (prompt *Prompt) Select(msg string, choices interface{}) (int, error) {
 	choice := prompt.Input(fmt.Sprintf("(%d-%d): ", 1, choicesLen))
 	i, err := strconv.Atoi(choice)
 	if err != nil {
-		return 0, err
+		return 0, errors.New("unable to parse selected input")
```
Comment (janisz) [inline pkg/prompt/prompt.go]:
```suggestion
		return 0, fmt.Errorf("unable to parse selected input: %s", err)
```

I'd like to have more context in errors

File: pkg/login/flow.go [hunk 2]
```diff
@@ -96,6 +96,13 @@ func (f *Flow) selectProvider(providers Providers) (*Provider, error) {
 	// Extract login provider candidates for implicit or manual selection.
 	var providerCandidates []*Provider
 	for _, provider := range providers.Slice() {
+		// If both username and password are passed without an explicit provider ID,
+		// we default to dcos-uid-password. Behind the scenes, dcos-uid-password-ldap
+		// might be used but the CLI does not take care of that.
+		if provider.ID == DCOSUIDPassword && f.flags.username != "" && f.flags.password != "" {
```
Comment (janisz) [inline pkg/login/flow.go]: Can you add a test for this?

**Comment by ArmandGrillet** [comment]
@mesosphere-ci run integration tests

**Comment by bamarni** [comment]
@mesosphere-ci run integration tests

**Pull request merged**